mindsdb/integrations/handlers/file_handler/file_handler.py

```diff
@@ -89,6 +90,42 @@ def query(self, query: ASTNode) -> Response:
                         error_message=f"Can't delete table '{table_name}': {e}",
                     )
             return Response(RESPONSE_TYPE.OK)
+
+        if type(query) is CreateTable:
+            # Check if the table already exists or if the table name contains more than one namespace
+            existing_files = self.file_controller.get_files_names()
+
+            if len(query.name.parts) != 1:
+                return Response(
+                    RESPONSE_TYPE.ERROR,
+                    error_message="Table name cannot contain more than one namespace",
+                )
+
+            table_name = query.name.parts[-1]
+            if table_name in existing_files:
+                return Response(
+                    RESPONSE_TYPE.ERROR,
+                    error_message=f"Table '{table_name}' already exists",
+                )
+
+            if query.is_replace:
+                self.file_controller.delete_file(table_name)
+
+            # Create a temp file to save the table
+            temp_dir_path = tempfile.mkdtemp(prefix="mindsdb_file_")
+            temp_file_path = os.path.join(temp_dir_path, f"{table_name}.csv")
+
+            # Create an empty file using with the columns in the query
+            df = pd.DataFrame(columns=[col.name for col in query.columns])
+            df.to_csv(temp_file_path, index=False)
+
+            self.file_controller.save_file(table_name, temp_file_path, file_name=f"{table_name}.csv")
+
+            # Remove the temp file created
+            shutil.rmtree(temp_dir_path, ignore_errors=True)
```

Comment:
If an error happens we don't remove and ignore the file? maybe we can use NamedTemporaryFile above that automatically handles cleanup?

Reply:
@ZoranPandovski Using `NamedTemporaryFile` does not work, because the FileController moves the file when saving it [here](https://github.com/mindsdb/mindsdb/blob/8118f35932ae1ec3c13b4afbc277a481a2eb0884/mindsdb/interfaces/file/file_controller.py#L107). Therefore, when the context manager looks to delete the file, it does not exist.
Instead, I added some error handling that I think achieves what you were getting at though.